### PR TITLE
#23 - #24 header, footer and hero desktop and mobile version - Emeka1993

### LINKED ISSUE #23: Create a responsive header for Mobile/Desktop
## Story

- **As a** website user 

- **I Want** to see the header with the navigation bar, hero and logo

- **So That** I can access different pages,

## Acceptance Criteria

1. Given I'm a website user
   - When I view the home page
   - Then I should see a nav bar and a hero image

2. Given I'm a website user 
   - When use a mobile phone 
   - Then it need be displayed as a hamburger menu 

## Implementation Notes

1. nav bar Links could be linked to real pages of CodeYourFuture (example donate page) 

image

### LINKED ISSUE #24: Create a responsive footer for Mobile/Desktop
## Story

- **As a** website user 

- **I want** to see the footer, social links and copyright.

- **So that** I can easy access the social links, legal and copyright information.

## Acceptance Criteria

1. Given a user
   - When I click a link in a footer 
   - Then i should be directed to the correct page.
2. Given a user
   - When access from the mobile phone
   - Then to be re-arrange in a mobile friendly way (Use the Figma design as a reference) 

image

### PR DESCRIPTION
**Pull request by Emeka1993**
Name your PR like this: ISSUE #23 #24 Emeka1993

## Description

Write a summary of your change. Explain why you have made it.

I have completed the footer, header, hero component and made it responsive on desktop and mobile.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #24 
Fixes #23 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-	<Routes>
-		<Route path="/" element={<Home />} />
-		<Route path="/graduates/:id" element={<Graduates />} />
-		<Route path="/about/this/site" element={<About />} />
-	</Routes>
-);
+const App = () => {
+  return (
+    <Layout>
+      <Hero
```
Comment (Luke-Rogerson) [inline client/src/App.js:11]: This component is nice and clean, but just bear in mind that this layout will now be used on every route. For example: you will see the hero image on all the sub-pages. Is that your intention?

File: client/src/App.js [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+        title="Hire a Graduate"
+      />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/graduates/:id" element={<Graduates />} />
+        <Route path="/about/this/site" element={<About />} />
```
Comment (Luke-Rogerson) [inline client/src/App.js:17]: You can probably just use the simpler path `/about`

File: client/src/Components/Footer.css [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  order: 2;
+  text-align: center;
+  font-family: "Helvetica";
+}
+
+.text {
+  font-size: 18px;
+}
+
+@media (max-width: 768px) {
```
Comment (Luke-Rogerson) [inline client/src/Components/Footer.css:44]: Nice and clean CSS, Emeka!

File: client/src/Components/Header.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      </div>
+      <div className="header-mobile">
+        <nav className="navbar-mobile">
+          <div className="navbar-mobile__logo">
+            <Link to="/"><img src={logo} alt="logo"></img></Link>
+          </div>
+          <div
+            className={`navbar-mobile__hamburger ${isOpen ? "open" : ""}`}
+            onClick={toggleNav}
+            onKeyDown={(e) => {
```
Comment (Luke-Rogerson) [inline client/src/Components/Header.js:35]: Cool that you even added keyboard support to the hamburger menu!

File: client/src/Components/Hero.css [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  width: 100%;
+
+}
+
+
+body {
+  margin: 0;
+}
+
+
```
Comment (Luke-Rogerson) [inline client/src/Components/Hero.css:16]: These blank lines can be deleted

File: client/src/App.js (hunk 1 above)
Comment (Emeka1993) [reply client/src/App.js:11]: Hi Luke, I did made some changes 2 days so I removed the hero from Layout!

File: client/src/Components/Footer.css (hunk 3 above)
Comment (Emeka1993) [reply client/src/Components/Footer.css:44]: Thank you very much!

**Pull request merged**